### PR TITLE
Engine Selection - added separate search and filter states for tabs + UX improvements

### PR DESCRIPTION
**Pull request by charliemantle**
- Exported `DiscoverySideBarContainerPure`
- Added separate search and filter states for each tab
- Added redux logic for fetching / handling engine categories
- Added async loading of engine categories to the category filter
- Made engine detail page a full page takeover
- Now clears all filters on tab switch
- Now resets filter active path on tab switch
- Removed categories filter title
- Changed default color of lozenge (push/pull pills)
- Added border between sidebar and content
- Filters now work on ‘Own Engines’ Page
- Added count around Your Engines Tab
- Search now works on both tabs
- Selected filters now persist on each tab
- Added new state when on "Own Engines" tab when there no filter results

### CODE REVIEW

File: packages/veritone-widgets/src/redux/modules/engineSelection/index.js [hunk 1]
```diff
@@ -294,7 +375,7 @@ export function initializeWidget(id) {
   };
 }
 
-export function refetchEngines(id) {
+export function refetchEngines(id, currentTabIndex) {
```
Comment (olslash) [inline packages/veritone-widgets/src/redux/modules/engineSelection/index.js]: why does this need an ID? relatedly, enginesModule.fetchEngines doesn't even use the ID you have it accepting...

File: packages/veritone-widgets/src/widgets/EngineSelection/EngineListView/SelectBar/SearchBar.js [hunk 2]
```diff
@@ -25,6 +26,24 @@ export default class SearchBar extends React.Component {
     searchQuery: ''
   };
 
+  state = {
+    searchQuery: this.props.searchQuery
+  };
+
+  static getDerivedStateFromProps(nextProps, prevState) {
```
Comment (olslash) [inline packages/veritone-widgets/src/widgets/EngineSelection/EngineListView/SelectBar/SearchBar.js]: this is sketchy pls read https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html and reconsider

File: packages/veritone-widgets/src/widgets/EngineSelection/EngineListView/SelectBar/SearchBar.js [hunk 3]
```diff
@@ -39,9 +58,10 @@ export default class SearchBar extends React.Component {
   };
 
   handleChange = event => {
-    if (event.target.value === this.props.searchQuery) {
-      return;
-    }
+    this.setState({
```
Comment (olslash) [inline packages/veritone-widgets/src/widgets/EngineSelection/EngineListView/SelectBar/SearchBar.js]: do you really need this state? why can't this be a normal controlled component?

File: packages/veritone-widgets/src/widgets/EngineSelection/EngineListView/SelectBar/index.js [hunk 4]
```diff
@@ -31,6 +32,10 @@ export default class SelectBar extends React.Component {
   };
 
   render() {
+    if (this.props.hideActionMenuItems) {
```
Comment (olslash) [inline packages/veritone-widgets/src/widgets/EngineSelection/EngineListView/SelectBar/index.js]: seems like "hide action menu items" actually means "hide this entire component", in which case, this hiding logic should be in the parent component that renders this one.

File: packages/veritone-widgets/src/widgets/EngineSelection/EngineListView/SelectBar/index.js [hunk 5]
```diff
@@ -41,46 +46,45 @@ export default class SelectBar extends React.Component {
...
+            onToggleSearch={this.props.onToggleSearch}
+            searchQuery={this.props.searchQuery}
+            isOpen={this.props.isSearchOpen}
+          />
+          {this.props.actionMenuItems && (
+            <div className={styles.actionMenuDividerContainer}>
+              <div className={styles.actionMenuDivider} />
+            </div>
+          )}
+          {this.props.actionMenuItems &&
```
Comment (olslash) [inline packages/veritone-widgets/src/widgets/EngineSelection/EngineListView/SelectBar/index.js]: try not to use `return` in these inner render functions (i wrote about this in the react standards doc)

also you shouldn't use index as a key... button text + icon class should be unique among items, right?

```javascript
{this.props.actionMenuItems &&
            this.props.actionMenuItems.map(
              ({ buttonText, iconClass, onClick }) =>
                buttonText ? (
                  <Button
                    key={`${buttonText}-${iconClass}`}
                    onClick={onClick}
                    color="primary"
                  >
                    {buttonText}
                  </Button>
                ) : (
                  <IconButton
                    key={`${buttonText}-${iconClass}`}
                    onClick={onClick}
                  >
                    <i className={iconClass} />
                  </IconButton>
                )
            )}
```

File: packages/veritone-react-common/src/components/Lozenge/index.js [hunk 6]
```diff
@@ -9,7 +9,7 @@ const Lozenge = ({ children, iconClassName, backgroundColor, textColor }) => {
     <div
       className={styles.lozenge}
       style={{
-        backgroundColor: backgroundColor || blue[500],
+        backgroundColor: backgroundColor || indigo[500],
```
[the remark this answers is not in the document]
Comment (charliemantle) [reply packages/veritone-react-common/src/components/Lozenge/index.js]: I made this change based on the feedback that it looks too much like a button using the default mui color.

File: packages/veritone-widgets/src/redux/modules/engineSelection/index.js (hunk 1 above)
Comment (charliemantle) [reply packages/veritone-widgets/src/redux/modules/engineSelection/index.js]: you're right, I added this by mistake, will fix.

**Pull request merged**